Content.Server/GameTicking/GameTicker.GameRule.cs

```diff
@@ -11,21 +11,38 @@ public partial class GameTicker
         [ViewVariables] private readonly HashSet<GameRulePrototype> _gameRules = new();
         public IEnumerable<GameRulePrototype> ActiveGameRules => _gameRules;
 
-        public bool AddGameRule(GameRulePrototype rule)
+        public void StartGameRule(GameRulePrototype rule)
+        {
+            if (!HasGameRule(rule))
+                EnableGameRule(rule);
+
+            RaiseLocalEvent(new GameRuleStartedEvent(rule));
+        }
+
+        public void EndGameRule(GameRulePrototype rule)
+        {
+            if (!HasGameRule(rule))
+                return;
+
+            DisableGameRule(rule);
+            RaiseLocalEvent(new GameRuleEndedEvent(rule));
+        }
```

Comment:
I think at the moment someone could `Disable` a game rule that has been started, which removes it from `_gameRules` without ever ending it?

I think in the same way that starting ensures that the rule has been enabled, disabling should ensure the rule has ended, but the end-function shouldn't necessarily call disable?

Also I think you might need to distinguish between has-rule and rule-started?